controllers/openstackcluster_controller.go

```diff
@@ -126,20 +126,22 @@ func (r *OpenStackClusterReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		if openStackCluster.Status.Bastion == nil {
 			openStackCluster.Status.Bastion = &infrav1.BastionStatus{}
 		}
-		changed, err := compute.ResolveReferencedMachineResources(scope, openStackCluster, &openStackCluster.Spec.Bastion.Instance, &openStackCluster.Status.Bastion.ReferencedResources)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		if changed {
+		if openStackCluster.Status.Bastion.ReferencedResources == nil {
```

Comment:
This is problematic because we adopt `ReferencedResources` once `OpenStackCluster.Status.Network != nil`.

I see multiple solutions:
* Keep `changed` but still convert the fields to pointers to follow API guidelines. This is the simplest option and I think easier to maintain. I'm fairly sure it's going to be working and the LOC change will be tiny.
* Add some logic here that adoption needs to be skipped if Status.Network is nil and in the next loops eventually Network will be created and Adoption will happen *once*. We wouldn't tolerate empty fields in this struct as this is required by the other resources. This is also a valid option, as we'll make adoption run **once**; but with the caveat that we have some weird logic (potentially fragile) where we require something before adoption can happen.

Reply:
@mdbooth if you can think of something else, I'm all ears :ear: :question: :thinking:

Reply:
Why don't we just skip this code block if network isn't set? If network isn't set then no bastion has ever been created, so there is also nothing to cleanup.